product_multi_ean/models/product_product.py

```diff
@@ -1,33 +1,46 @@
-# -*- coding: utf-8 -*-
 # © 2012-2014 Guewen Baconnier (Camptocamp SA)
 # © 2015 Roberto Lizana (Trey)
 # © 2016 Pedro M. Baeza
+# © 2018 Xavier Jimenez (QubiQ)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from openerp import models, fields, api, _
-from openerp.exceptions import Warning as UserError
-from openerp.addons.product import product as addons_product
+from odoo import models, fields, api, _
+from odoo.exceptions import Warning as UserError
 
 
 class ProductEan13(models.Model):
     _name = 'product.ean13'
     _description = "List of EAN13 for a product."
     _order = 'sequence, id'
 
-    name = fields.Char(string='EAN13', size=13, required=True)
-    sequence = fields.Integer(string='Sequence', default=0)
+    name = fields.Char(
+        string='EAN13',
+        size=13,
+        required=True,
+    )
+    sequence = fields.Integer(
+        string='Sequence',
+        default=0,
+    )
     product_id = fields.Many2one(
-        string='Product', comodel_name='product.product', required=True)
+        string='Product',
+        comodel_name='product.product',
+        required=True,
+    )
 
     @api.multi
     @api.constrains('name')
     @api.onchange('name')
     def _check_name(self):
+        barcode_obj = self.env['barcode.nomenclature']
+
         for record in self:
-            if not addons_product.check_ean(record.name):
-                raise UserError(
-                    _('You provided an invalid "EAN13 Barcode" reference. You '
-                      'may use the "Internal Reference" field instead.'))
+            if record.filtered('name'):
```

Comment:
This should be put in the previous line: `for record in self.filtered('name'):` The idea is to save one line and one cyclomatic complexity level